library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/AbstractPojoJoin.java

```diff
@@ -220,8 +233,24 @@ protected void setObjectForResult(Map<String,PojoUtils.Getter> stream, Object in
         }
         for (Object lObj:left) {
           for (Object rObj:right) {
-            setObjectForResult(leftGettersStream, lObj,o);
-            setObjectForResult(rightGettersStream, rObj,o);
+            if (outputToInputMap != null) {
```

Comment:
Not always as this may be a scenario when user wants to add all the fields of output POJO and the mapping is based on field names.